kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/config/AbstractKubeConfigLoader.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.config;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.io.ResourceResolver;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Abstract implementation of {@link KubeConfigLoader} which provides common methods for loading kube config.
+ */
+@Internal
+public abstract class AbstractKubeConfigLoader implements KubeConfigLoader {
```

Comment:
does this need to be public? Can it be sealed?

Reply:
It has to be `public` since it needs to be extended when a kube config has to be loaded from a cloud service. Can't be `sealed` since we don't implement loading of kube config from cloud services.

Reply:
ok should it still be considered `@Internal` in that case?

Reply:
Removed `@Internal`